### ISSUE TITLE
nested relations ignored if parent is eager

### ISSUE THREAD

**Issue by ilovett** [typeorm/typeorm#5169]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```ts
        const example = await Example.findOne(
          {  id  },
          { relations: ["parent", "parent.childOne", "parent.childTwo", "other"] }
        );
```

In this example, if `Example` entity's `parent` relation is eager: true, then `childOne` and `childTwo` will not be loaded as requested.

```ts
  @ManyToOne(
    () => Parent,
    (parent: Parent) => parent.example,
    {
      eager: true // disabling this and the above code will load nested relations
    }
  )
  @JoinColumn()
  parent: Parent | null;
```

**Comment by ardyfeb** [comment]
@pleerock

**Comment by olivermrbl** [comment]
Anything new on this bug? :)

**Comment by R3D13** [comment]
Also waiting for a statement.
This and #5684 seem to be related and pr #5685 might even fix both issues.

**Comment by satanTime** [comment]
Yep, this is what is fixed by #5685. I've rebased PR.
